borsh-derive/src/internals/deserialize/mod.rs

```diff
@@ -0,0 +1,8 @@
+mod r#enum;
+mod r#struct;
+mod union;
+
+pub use r#enum::enum_de;
+use r#struct::field_deserialization_output;
+pub use r#struct::struct_de;
```

Comment:
I would avoid naming collision with the keywords and just name the modules in plural form (`enums`, `structs`, `unions`) or come up with something else, as this raw-syntax attracts too much attention for no reason.

Reply:
Plural `structs` is concise and great compared to `struct_`. Hasn't come up my mind compared to `r#struct`.